assets/src/edit-story/components/dialog/dialog.js

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { ThemeProvider } from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import {
+  Dialog as StyledDialog,
+  Button,
+  BUTTON_SIZES,
+  BUTTON_TYPES,
+  theme,
+  lightMode,
+} from '../../../design-system';
+
+const Dialog = ({
+  open,
+  onClose,
+  closeText,
+  onConfirm,
+  confirmText,
+  actions,
+  children,
+  ...rest
+}) => {
+  return (
+    <ThemeProvider theme={{ ...theme, colors: lightMode }}>
+      <StyledDialog
+        isOpen={open}
+        onClose={onClose}
+        {...rest}
+        actions={
+          actions || (
```

Comment:
If `actions` are passed in to dialog allow them to override the default structure. This is for cases where there is 1 button or maybe the cancel button does something different than just closing it, or the confirmation button is actually a link.

Reply:
I'd rather like to see two versions of the component here to make the API and prop types more consistent. A basic version (`<Dialog />`?), that just takes an `actions` object, and another specialized one (`<ConfirmDialog />`?) that takes `closeText`, `onConfirm`, and `confirmText` and passes that as a structured `actions` object to the underlying basic `<Dialog />`. WDYT?

Reply:
Alternatively (looking through the usages), it could also be generalized as:

```js
          actions={(
            <>
              <Button
                type={BUTTON_TYPES.TERTIARY}
                size={BUTTON_SIZES.SMALL}
                onClick={onSecondary || onClose}
                {...secondaryProps}
              >
                {secondaryText}
              </Button>
              <Button
                type={BUTTON_TYPES.PRIMARY}
                size={BUTTON_SIZES.SMALL}
                onClick={onPrimary}
                {...primaryProps}
              >
                {primaryText}
              </Button>
            </>
          )}
```

Then all the cases could be solved by providing the right mix of these properties, I believe?

Reply:
Hm, except for the `termsDialog`, that only has one button... Maybe it can depend on the text being non-empty?

```js
          actions={(
            <>
              {secondaryText && (<Button
                type={BUTTON_TYPES.TERTIARY}
                size={BUTTON_SIZES.SMALL}
                onClick={onSecondary || onClose}
                {...secondaryProps}
              >
                {secondaryText}
              </Button>)}
              {primaryText && (<Button
                type={BUTTON_TYPES.PRIMARY}
                size={BUTTON_SIZES.SMALL}
                onClick={onPrimary}
                {...primaryProps}
              >
                {primaryText}
              </Button>)}
            </>
          )}
```

But it is becoming a bit much. Not sure what I prefer here, really. Could also be `<OneButtonDialog />` and `<TwoButtonDialog />`.

Reply:
Hm, feels a bit too much having separate components matching exactly except for 1 vs 2 buttons.

@barklund Would it be more clear if we'd basically just rename the props and accept only `primaryBtnProps` and `secondaryBtnProps` in the `Dialog` instead? And then do something like this in the component (similar to what you suggested above but the `Dialog` accepting button props as one object for each button):

```js

const { text: primaryText, ...primaryProps } = primaryBtnProps;
const { text: secondaryText, ...secondaryProps } = secondaryBtnProps;

actions={(
  <>
    {primaryText && (<Button
      type={BUTTON_TYPES.TERTIARY}
      size={BUTTON_SIZES.SMALL}
      {...secondaryProps}
    >
      {secondaryText}
    </Button>)}
    {secondaryText && (<Button
      type={BUTTON_TYPES.PRIMARY}
      size={BUTTON_SIZES.SMALL}
      {...primaryProps}
    >
      {primaryText}
    </Button>)}
  </>
)}

```

An alternative could also do something like this for readability:
`actions={actions || getDefaultButtons(primaryBtnProps, secondaryBtnProps)}`

... and then use the buttons above in `getDefaultButtons`.

(This would still leave the option to pass in `actions` directly, too. To me, having the 2 button dialogue as the default and using `actions` in case of the 1 button dialogue also makes sense.)

Reply:
Yeah, the 1-vs-2 button dialog was also a more extreme suggestion. I prefer some of the other ones I made.

Either of your options sounds reasonable to me as well, @miina (except for the swapped boolean checks 😂 ).

On the helper function variant, I'd prefer to use an actual component, but other than that, works for me:

```js
actions={actions || <DefaultButtons primary={primary} secondary={secondary} />}
```

Reply:
ok, @barklund - spent a little time on this and think it's at a good point of compromise. lemme know what you think. 

I gutted the terminology of "close" and "confirm" in favor of secondary and primary. When those `primaryText` and/or `secondaryText` are present, it renders the respective buttons.  There's still the `actions` escape hatch just in case it's needed in the future. 

This drastically simplifies the API but keeps the logic out of the base dialog in the design system. 

Thoughts?